sh_scrapy/middlewares.py

```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 import itertools
+from warnings import warn
```

Comment:
This is for the call in `HubstorageDownloaderMiddleware.from_crawler()`, added in some older change.